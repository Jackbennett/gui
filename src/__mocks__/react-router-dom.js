import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
// eslint-disable-next-line no-import-assign
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
