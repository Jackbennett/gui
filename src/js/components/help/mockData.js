export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
