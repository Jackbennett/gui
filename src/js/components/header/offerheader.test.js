import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import OfferHeader from './offerheader';
import { undefineds } from '../../../../tests/mockData';

describe('DeviceNotifications Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(
      <MemoryRouter>
        <OfferHeader />
      </MemoryRouter>
    );
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
