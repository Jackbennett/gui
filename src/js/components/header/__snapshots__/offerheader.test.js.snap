// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceNotifications Component renders correctly 1`] = `
<div
  class="offerBox"
  id="offerHeader"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    style="margin-right: 2px; height: 16px; vertical-align: bottom;"
    viewBox="0 0 24 24"
  >
    <path
      d="M21.41 11.58l-9-9C12.05 2.22 11.55 2 11 2H4c-1.1 0-2 .9-2 2v7c0 .55.22 1.05.59 1.42l9 9c.36.36.86.58 1.41.58.55 0 1.05-.22 1.41-.59l7-7c.37-.36.59-.86.59-1.41 0-.55-.23-1.06-.59-1.42zM5.5 7C4.67 7 4 6.33 4 5.5S4.67 4 5.5 4 7 4.67 7 5.5 6.33 7 5.5 7z"
    />
  </svg>
  <span>
    Try out the new 
    <b>
      Remote terminal
    </b>
     feature for free on your current plan until March 31st.Â 
  </span>
  <a
    href="https://docs.mender.io/add-ons/remote-terminal"
    rel="noreferrer"
    target="_blank"
  >
    Read the documentation to learn how to enable it
  </a>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root"
    focusable="false"
    style="margin-left: 4px; height: 16px; cursor: pointer;"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
    />
  </svg>
</div>
`;
